### PR TITLE
Better custom `Debug` implementations

### LINKED ISSUE #63: Revisit `Debug` implementations
We should do a pass over the existing `Debug` implementations so they appear nicer in tracing output.

For instance, instead of
```
BlockHeaderHash([247, 3, 134, 210, 114, 134, 208, 233, 93, 156, 161, 56, 104, 238, 53, 170, 136, 70, 17, 174, 178, 248, 161, 167, 29, 125, 152, 0, 0, 0, 0, 0])
```
we could have a custom `Debug` impl that uses `hex` to do, e.g.,
```
BlockHeaderHash(81cd02ab7e569e8bcd9317e2fe99f2de44d49ab2b8851ba4a308000000000000)
```

This also allows using tracing filters on hash values, etc.

### PR DESCRIPTION
**Pull request by dconnolly**
Resolves #63

### CODE REVIEW

**Comment by dconnolly** [comment]
This is enough impls for now.

File: zebra-chain/Cargo.toml [hunk 1]
```diff
@@ -11,5 +11,6 @@ edition = "2018"
 thiserror = "1"
 byteorder = "1.3"
 chrono = "0.4"
-#hex = "0.4" This conflicts with tracing-subscriber?
+failure = "0.1"
```
Comment (hdevalence) [inline zebra-chain/Cargo.toml]:
```suggestion
```
did this come from merging or does this branch use `failure`?

**Review by hdevalence — APPROVED** [review]
LGTM!!

File: zebra-chain/Cargo.toml (hunk 1 above)
Comment (hdevalence) [reply zebra-chain/Cargo.toml]: looks like this was a merge artifact -- I don't see `failure` used anywhere

**Pull request merged**